hivemq4/base-image/Dockerfile

```diff
@@ -37,27 +36,35 @@ RUN set -x \
         && { command -v gpgconf && gpgconf --kill all || :; } \
         && chmod +x /usr/local/bin/gosu \
         && gosu nobody true \
-        && apt-get purge -y gpg dirmngr && rm -rf /var/lib/apt/lists/* \
-        && mkdir -p /docker-entrypoint.d
+        && apt-get purge -y gpg dirmngr && rm -rf /var/lib/apt/lists/*
+
+
+#Additional build image to unpack the zip file and change the permissions without retaining large layers just for those operations
+FROM javabase as unpack
+
+ARG HIVEMQ_VERSION=4.0.0
```

Comment:
We have this twice in the file.

Reply:
can we move all ARGs to the top of the file?

Reply:
> An ARG instruction goes out of scope at the end of the build stage where it was defined. To use an arg in multiple stages, each stage must include the ARG instruction.

Reply:
```suggestion
ARG HIVEMQ_VERSION
```